compiler/rustc_borrowck/src/region_infer/mod.rs

```diff
@@ -2000,8 +2000,14 @@ impl<'tcx> RegionInferenceContext<'tcx> {
         let cause_code = path
             .iter()
             .find_map(|constraint| {
-                if let ConstraintCategory::Predicate(def_id, predicate_span) = constraint.category {
-                    Some(ObligationCauseCode::BindingObligation(def_id, predicate_span))
+                if let ConstraintCategory::Predicate(predicate_span) = constraint.category {
+                    // We currentl'y doesn't store the `DefId` in the `ConstraintCategory`
+                    // for perforamnce reasons. The error reporting code used by NLL only
+                    // uses the span, so this doesn't cause any problems at the moment.
+                    Some(ObligationCauseCode::BindingObligation(
+                        CRATE_DEF_ID.to_def_id(),
+                        predicate_span,
+                    ))
```

Comment:
We should create a new ObligationCauseCode variant instead, but it's fine to do this I guess.

Reply:
I was trying to mimic the non-NLL code, which also uses `ObligationCauseCode::BindingObligation` for this message:

https://github.com/rust-lang/rust/blob/2b6ed3b675475abc01ce7e68bb75b457f0c85684/compiler/rustc_infer/src/infer/error_reporting/note.rs#L396-L405